packages/component/src/provider.ts

```diff
@@ -263,9 +263,15 @@ export class Provider<T extends Name> {
    *
    * @returns a function to unregister the callback
    */
-  onInit(callback: OnInitCallBack<T>): () => void {
+  onInit(callback: OnInitCallBack<T>, identifier?: string): () => void {
```

Comment:
Can you please add a comment for the argument `identifier`?

Reply:
Should we trigger the callback for all instances that have already been created? That's what would happen if those were created after the onInit callback is registered. So that we have a consistent behavior no matter whether the callback is registered before or after initialization (it will be invoked for every instances created).

Reply:
As discussed offline, I think the current behavior already achieves this. All callbacks that are registered for an existing component will be invoked once when registered (new logic). All callbacks that are registered before a component is available will be invoked once the component is available (existing logic). 

Let me know if there are any more questions.

Reply:
Sorry I wasn't clear. What I meant is we don't need the `identifier` argument, and a callback can be invoked multiple times (once per identifier) when registered if the component support multiple instances and there are already multiple instances initialized.

Reply:
as per discussion offline, we will change the callback to be associated with an identifier.